### ISSUE TITLE
modem: stuck while quitting

### ISSUE THREAD

**Issue by mpvader** [victronenergy/venus#325]
Happened on an issue where the modem disappears every ~1m30s. I'll make another issue for that.

```
2018-08-28 13:25:41.015851500 INFO     Loglevel set to INFO
2018-08-28 13:25:41.027942500 INFO     Starting dbus-modem 0.2 on /dev/ttyUSB2 at 115200 bps
2018-08-28 13:25:41.115382500 INFO     registered ourselves on D-Bus as com.victronenergy.modem
2018-08-28 13:25:41.315424500 INFO     Waiting for localsettings
2018-08-28 13:25:41.482265500 INFO     Waiting for modem to become ready
2018-08-28 13:25:46.905127500 INFO     Modem ready
2018-08-28 13:27:11.989911500 ERROR    Write error, quitting
2018-08-28 13:27:12.178254500 ERROR    Read error, quitting
(stuck, for several minutes, and then I killed it)
```

Note that I've seen the same thing earlier on another installation, but then the other way around, first a write error and then a read error:
```
2018-06-06 20:51:47.572027500 DEBUG    < OK
2018-06-06 20:51:52.554003500 DEBUG    > AT+CGSETV=44,0,0
2018-06-06 20:51:52.564437500 DEBUG    < AT+CGSETV=44,0,0
2018-06-06 20:51:52.564446500 DEBUG    < OK
2018-06-06 20:51:57.561410500 DEBUG    > AT+CGSETV=44,1,0
2018-06-06 20:51:57.573600500 ERROR    Read error, quitting
2018-06-06 20:51:57.573610500 DEBUG    Stopping pppd
2018-06-06 20:51:57.573612500 ERROR    Write error, quitting
2018-06-06 20:51:57.597736500 DEBUG    Stopping pppd
```

**Comment by mpvader** [comment]
Afterwards I noticed this:
```
root@ccgx:~# ps | grep modem
 1453 root      1580 S    supervise dbus-modem.ttyUSB2
 1511 root      1592 S    multilog t s99999 n8 /log/dbus-modem.ttyUSB2
 5210 root      3060 S    {start-modem.sh} /bin/bash /opt/victronenergy/dbus-modem/start-modem.sh ttyUSB2
 5213 root     25700 S    {dbus-modem.py} /usr/bin/python -u /opt/victronenergy/dbus-modem/dbus-modem.py -s /dev/ttyUSB2
 5542 root      1580 S    supervise dbus-modem.ttyUSB3
 5599 root      1724 S    multilog t s99999 n8 /log/dbus-modem.ttyUSB3
 6970 root      3060 S    {start-modem.sh} /bin/bash /opt/victronenergy/dbus-modem/start-modem.sh ttyUSB3
 6973 root      7528 R    {dbus-modem.py} /usr/bin/python -u /opt/victronenergy/dbus-modem/dbus-modem.py -s /dev/ttyUSB3
 6978 root      2248 S    grep modem
```

Could be that the port is still in use by the hung script.

**Comment by mpvader** [comment]
Fixed in v2.20~58
